src/movies.js

```diff
@@ -1,16 +1,113 @@
 // Iteration 1: All directors? - Get the array of all directors.
+function getAllDirectors(movies){
+
+const directorF =  movies.map(function(movie){
+        return movie.director
+    })
+
+return directorF
+}
+
+
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
+
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
 
+function howManyMovies(movies){
+    let drama;
+    let spiel = movies.filter((movie) => {
+    for (let i=0; i < movie.genre.length; i++){
+       drama = movie.genre[i] == "Drama" && movie.director == "Steven Spielberg";
+       if(drama){break;}
+    } 
+    return drama
+}) 
+return spiel.length;
+};
+
+function howManyMovies(movies){
+    let Drama = movies.filter(function(ele){
+      return ele.genre.includes("drama")
+    })
+    return Drama;
+  }
+
+
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
 
+function ratesAverage(movies){
+  
+    let sum = movies.reduce(function(acc,value){
+      return acc + value.rate
+    },0)
+    return sum/sum.length;
+  }
+
 // Iteration 4: Drama movies - Get the average of Drama Movies
+function dramaMoviesRate(movies){
+    let s = movies.filter(function(elw){
+      return elw.genre.includes("drama")
+    })
+    return rateAverage(s);
+    }
 
 // Iteration 5: Ordering by year - Order by year, ascending (in growing order)
+function orderByYear(movies){
+
+   
+    let clone = JSON.parse(JSON.stringify(movies))
+    
+    let order =clone.sort(function(a,b){
+      if( a.year > b.year){
+        return 1;
+      }else if( a.year < b.year){
+        return -1;
+      }else{
+        return 0;
```

Comment:
To pass the test to order alphabetically if they have the same year, instead of returning 0 here you can add a new conditional that checks their title.